content/docs/user-guide/scripting/script-canvas/programmer-guide/custom-nodes/custom-node-definition-reference.md

```diff
@@ -0,0 +1,382 @@
+
+Functions can use the [Parameter](#propertydata) element to specify data inputs.
+
+```xml
+<Function Name="RandomColor">
+    <Parameter Name="MinValue" DefaultValue="Data::ColorType(0.0f, 0.0f, 0.0f, 1.0f)"/>
+    <Parameter Name="MaxValue" DefaultValue="Data::ColorType(1.0f, 1.0f, 1.0f, 1.0f)"/>
+</Function>
+
+<Function Name="IsValidFindPosition"> <!-- Used by ContainsString to evaluate the condition to branch -->
+    <Parameter Name="Position" Description="The string find position to check against"/>
+</Function>
+
+<Function Name="ContainsString" Branch="IsValidFindPosition" BranchWithValue="True">
+    <Parameter Name="Source" Description="The string to search in."/>
+    <Parameter Name="Pattern" Description="The substring to search for."/>
+    <Parameter Name="Search From End" Description="Start the match checking from the end of a string."/>
+    <Parameter Name="Case Sensitive" Description="Take into account the case of the string when searching."/>
+</Function>
+
+```
+
+Functions can use the __Out__ element to specify the name for each of the [Branch](#branch) results.
+
+__NOTE:__ There may only be 2 __Out__ elements for branching functions, if more than 2 exist an error will be displayed during compilation.
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Name__ | Required | The name of the function | `<Function Name="Clamp">` |
+
+```xml
+<Function Name="BranchExample" Branch="IsPositive" BranchWithValue="True">
+    <Out Name="Is Positive"/>
+    <Out Name="Is Negative"/>
+</Function>
+```
+----
+
+## Library
+
+Library is used to create a collection of functions
+
+| __Attribute__   |  __Requirements__  | __Description__ | __Example__ |
+| ---------- | ------------| ------------| ------------| 
+| __Include__ | Required | The relative path to the C++ include file that will contain the library function declarations | `Include="Include/ScriptCanvas/Libraries/Math/Color.h"` |
+| __Namespace__ | Required | The namespace that holds the C++ functions in this library | `Namespace="ScriptCanvas::ColorFunctions"` |
+| __Category__ | Recommended | Category will define where this node will be places on the Node Palette, you can nest categories by separating with the / character (e.g. "My Nodes/Math") | `Category="Math/Color"` |
+
+
+```xml
+<Library Include="Include/ScriptCanvas/Libraries/Math/MathFunctions.h"
+        Namespace="ScriptCanvas::MathRandoms"
+        Category="Math/Random">
+
+    <!-- Function Definitions -->
+
+</Library>
+```
+----
+
+## <a href="#grammar"></a> Grammar
+
+The Script Canvas Grammar nodes are nodes that have a direct translation into the backend's execution format, currently this is the Lua language. It is not generally required to create new grammar nodes if the Script Canvas Nodeable format is sufficient, but in some situations it may be desired to extend the Script Canvas grammar.
```

Comment:
No need to use "currently" when there's no guarantee if and when this will change. The documentation should be updated if this changes.
```suggestion
Grammar nodes are nodes that have a direct translation into the backend's execution format, which is Lua. You do not need to create new grammar nodes if the Script Canvas Nodeable format is sufficient. However, in some situations it might be desirable to extend the Script Canvas grammar.
```